### ISSUE TITLE
[COGNITION] Smooth scrolling SHOULD be stopped when reduced motion is preferred

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#6701]
## Description

To improve accessibility for all users, we **SHOULD** respect the accessibility preference to reduce motion option when set.

When enabled, we need to stop all smooth scrolling animation.
When disabled, smooth scrolling _should_ be okay.

## Acceptance Criteria

- Smooth scrolling is disabled when the OS reduce motion accessibility option is set.

## Environment

* Operating System: All
* Browser: All
* Screenreading device: N/A
* Server destination: All

## Steps to Recreate

1. Log in and open _any_ form
1. On a page where user input is expected, press the "Continue" button.
1. The browser will show an error and scroll the page to place the error at the top of the page
1. Or pressing "Continue" on a valid page or "Back" will also smooth scroll to bring the `h1` to the top of the page.

## Possible Fixes (optional)

Since we're using the React scroll library to add scroll animation, adding a css media query to stop css animation will not suffice. Instead, a scroll duration of zero should be passed to the library  

Use a JavaScript method to test media query settings: 

```js
// matchMedia will return true (when OS reduce motion accessibility option is set)
const duration = window?.matchMedia("(prefers-reduced-motion: reduce)")?.matches
  ? 0
  : 500; // milliseconds
```

## WCAG or Vendor Guidance (optional)

* [WCAG animation from interactions success criterion](https://www.w3.org/WAI/WCAG21/Understanding/animation-from-interactions.html)
* [MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Testing_media_queries)

**Comment by jenstrickland** [comment]
It might help to have a conversation with @1Copenut and me, and possibly bring to the design system design council. Prefers reduced motion is great for those who are aware of it and have it available as an option. For those with vestibular conditions who don't know, the smooth scroll may still be an issue, and the guidance from Level Access is not to use it at all.

**Comment by joshkimux** [comment]
[Opening a DST issue and closing this out](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1604)
